### DISCUSSION TITLE
Understanding AWS CloudWatch Billing?

### DISCUSSION THREAD

**Discussion by fex01** [infracost/infracost#2727] [Q&A]
Hi there,

I'm trying to understand my AWS bill by comparing the AWS Billing overview with infracost output. Most items / cost components match - but CloudWatch leaves me a bit confused.

The AWS Billing Overview list the three factors `metric-month`, `requests` and `PutLogEvents` (measured in GBs).
In my infracost breakdown, CloudWatch seems to match with resource type `aws_cloudwatch_log_group`:

```json
"name": "module.vpc.aws_cloudwatch_log_group.flow_log[0]",
"resourceType": "aws_cloudwatch_log_group",
"tags": {
  "LAB": "thesis",
  "infra": "terraform"
},
"metadata": {},
"costComponents": [
  {
    "name": "Data ingested",
    "unit": "GB",
    "hourlyQuantity": null,
    "monthlyQuantity": null,
    "price": "0.5985",
    "hourlyCost": null,
    "monthlyCost": null
  },
  {
    "name": "Archival Storage",
    "unit": "GB",
    "hourlyQuantity": null,
    "monthlyQuantity": null,
    "price": "0.0315",
    "hourlyCost": null,
    "monthlyCost": null
  },
  {
    "name": "Insights queries data scanned",
    "unit": "GB",
    "hourlyQuantity": null,
    "monthlyQuantity": null,
    "price": "0.0059",
    "hourlyCost": null,
    "monthlyCost": null
  }
]
```

But how do `metric-month`, `requests` and `PutLogEvents` match with "Data ingested", "Archival Storage" and "Insights queries data scanned"??

Or is there no matching resource type for CloudWatch and `aws_cloudwatch_log_group` stands for something else? At least I can't find any other item in the AWS Billing Overview that could match this resource type 🤔 

Sorry for the confused question - and thx in advance :-)
Felix

**Comment by alikhajeh1** [comment]
@fex01 great question! It's been a while since I looked into that code, I'll dig into it and get back to you in the next few days :)

**Comment by aliscott** [comment]
@fex01 here's what I could find:

* `metric-month` will be the value in the 'Metrics' [tab here](https://aws.amazon.com/cloudwatch/pricing/). Infracost tracks metric costs for EC2 instances against the `aws_instance` resource as 'EC2 detailed monitoring'.
* I think `requests` will also be the `GetMetricData` prices listed in the 'Metrics' [tab here](https://aws.amazon.com/cloudwatch/pricing/). Infracost currently doesn't track this.
* `PutLogInvents` is the same as 'Data ingested', which Infracost tracks against the `aws_cloudwatch_log_group` resource. You can see the prices for the logging in the 'Logs' [tab here](https://aws.amazon.com/cloudwatch/pricing/). This will be the same as 'Collect (Data Ingestion) ' on that page.
